src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -720,6 +727,15 @@ private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, Dat
                 (_evaluationStep, _) = AddEvaluationStep(_softMaxTensor, _labelTensor);
                 _softmaxTensorName = _softMaxTensor.name;
                 _predictionTensorName = _prediction.name;
+
+                VBuffer<ReadOnlyMemory<char>> keysVBuffer = default;
+                inputSchema[labelColumnName].GetKeyValues(ref keysVBuffer);
```

Comment:
>GetKeyValues [](start = 45, length = 12)

You need to check whether the label column has key values first.

Reply:
Since it is possible that `_keyValueAnnotations` will be null, you need to account for that at serialization/deserialization time.

---
In reply to: [328528099](https://github.com/dotnet/machinelearning/pull/4228#discussion_r328528099) [](ancestors = 328528099)

Reply:
Label column can be KeyType, uint32, long but for now just put an assert that label column is keytype and open an issue to make label column be keytype, uint32, long.

---
In reply to: [329080695](https://github.com/dotnet/machinelearning/pull/4228#discussion_r329080695) [](ancestors = 329080695,328528099)

Reply:
Key values annotations can be null even if the column is a key type.

---
In reply to: [329333010](https://github.com/dotnet/machinelearning/pull/4228#discussion_r329333010) [](ancestors = 329333010,329080695,328528099)